### ISSUE TITLE
Pulsar-client 2.5.0 throws UnsatisfiedLinkError when snappy is used for compression¥

### ISSUE THREAD

**Issue by feeblefakie** [apache/pulsar#6260]
**Describe the bug**
UnsatisfiedLinkError is thrown when snappy compression is used.

The actual error:
```
java.lang.UnsatisfiedLinkError:
org.apache.pulsar.shade.org.xerial.snappy.SnappyNative.maxCompressedLength(I)I
```

The error won't happen if compression is not used or other compression like LZ4 is used.
Also, the error won't happen if `pulsar-perf` that comes with the binary is used with SNAPPY specified.

**To Reproduce**
1. Create a producer with pulsar-client 2.5.0
```
Producer<byte[]> producer = client.newProducer().compressionType(CompressionType.SNAPPY).topic(topic).create();
```

**Expected behavior**
Just work with errors.

**Screenshots**
None.

**Desktop (please complete the following information):**
 - OS: Mac OSX

**Additional context**
My guess:
The binary has `org.xerial.snappy-snappy-java-1.1.1.3.jar` which has snappy `.so` files, but if I use pulsar-client those `.so` files are not installed, so it is producing such errors ?

**Comment by yjshen** [comment]
I'm trying to reproduce the bug, using:

```java
public static void main(String[] args) throws PulsarClientException, InterruptedException {
		PulsarClient c = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build();
		String topic = "zzz";

		Producer<byte[]> a = c.newProducer().topic(topic).compressionType(CompressionType.SNAPPY).create();
		a.send("1".getBytes());  // 
		a.send("b".getBytes());

		Reader r = c.newReader().topic(topic).startMessageId(MessageId.earliest).create();
		System.out.println(new String(r.readNext().getData()));
		System.out.println(new String(r.readNext().getData()));

	}
```

Snappy hang at `a.send(1)` forever, however zlib, lz4, ZSTD works as expected.

**Comment by yjshen** [comment]
When I change the artifact to `pulsar-client-original`, the snappy works for the code above.

**Comment by yjshen** [comment]
Fixed in #6375
